Signal/src/ViewControllers/NewGroupViewController.m

```diff
@@ -66,243 +79,353 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder
+
+    const CGFloat kAvatarSize = 68.f;
+    UIImageView *avatarView = [UIImageView new];
+    _avatarView = avatarView;
+    avatarView.layer.borderColor = UIColor.clearColor.CGColor;
+    avatarView.layer.masksToBounds = YES;
+    avatarView.layer.cornerRadius = kAvatarSize / 2.0f;
+    avatarView.contentMode = UIViewContentModeScaleAspectFill;
+    [threadInfoView addSubview:avatarView];
+    [avatarView autoVCenterInSuperview];
+    [avatarView autoPinEdgeToSuperviewEdge:ALEdgeLeft];
+    [avatarView autoSetDimension:ALDimensionWidth toSize:kAvatarSize];
+    [avatarView autoSetDimension:ALDimensionHeight toSize:kAvatarSize];
+    [self updateAvatarView];
+
+    UITextField *groupNameTextField = [UITextField new];
+    _groupNameTextField = groupNameTextField;
+    groupNameTextField.textColor = [UIColor blackColor];
+    groupNameTextField.font = [UIFont ows_dynamicTypeTitle2Font];
+    groupNameTextField.placeholder = NSLocalizedString(@"NEW_GROUP_NAMEGROUP_REQUEST_DEFAULT", @"");
+    groupNameTextField.delegate = self;
+    [groupNameTextField addTarget:self
+                           action:@selector(groupNameDidChange:)
+                 forControlEvents:UIControlEventEditingChanged];
+    [threadInfoView addSubview:groupNameTextField];
+    [groupNameTextField autoVCenterInSuperview];
+    [groupNameTextField autoPinEdge:ALEdgeLeft toEdge:ALEdgeRight ofView:avatarView withOffset:16.f];
+    [groupNameTextField autoPinEdgeToSuperviewEdge:ALEdgeRight withInset:16.f];
+
+    [avatarView addGestureRecognizer:[[OWSAnyTouchGestureRecognizer alloc] initWithTarget:self
+                                                                                   action:@selector(avatarTouched:)]];
+    avatarView.userInteractionEnabled = YES;
+
+    return firstSectionHeader;
+}
+
+- (void)avatarTouched:(UIGestureRecognizer *)sender
+{
+    if (sender.state == UIGestureRecognizerStateRecognized) {
+        [self showChangeGroupAvatarUI:nil];
     }
 }
 
-- (BOOL)isContactHidden:(Contact *)contact
+#pragma mark - Table Contents
+
+- (void)updateTableContents
 {
-    if (contact.parsedPhoneNumbers.count < 1) {
-        // Hide contacts without any valid phone numbers.
-        return YES;
+    OWSTableContents *contents = [OWSTableContents new];
+
+    __weak NewGroupViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
+
+    NSArray<SignalAccount *> *signalAccounts = self.contactsViewHelper.signalAccounts;
+    NSMutableSet *nonContactMemberRecipientIds = [self.memberRecipientIds mutableCopy];
+    for (SignalAccount *signalAccount in signalAccounts) {
+        [nonContactMemberRecipientIds removeObject:signalAccount.recipientId];
     }
 
-    if ([self isCurrentUserContact:contact]) {
-        // We never want to add ourselves to a group.
-        return YES;
+    // Non-contact Members
+
+    if (nonContactMemberRecipientIds.count > 0 || signalAccounts.count < 1) {
```

Comment:
edit: disregard my previous comment - I see now why this is different for edit vs. create. 👍